scripts/install-tests.sh

```diff
@@ -19,6 +19,41 @@ else
     pip3 install --user -r https://raw.githubusercontent.com/dockstore/dockstore/develop/dockstore-webservice/src/main/resources/requirements/1.7.0/requirements3.txt
 fi
 
+if [ "${TESTING_PROFILE}" = "singularity-tests" ]; then
+    # Install singularity from source
+    # need singularity > 3.0.0, which is not available as an ubuntu package
+    sudo apt-get update && sudo apt-get install -y \
+    build-essential \
+    libssl-dev \
+    uuid-dev \
+    libgpgme11-dev \
+    squashfs-tools \
+    libseccomp-dev \
+    pkg-config
+
+    # Install Go (needed to install singularity)
+    export VERSION=1.11 OS=linux ARCH=amd64 && \
+    wget https://dl.google.com/go/go$VERSION.$OS-$ARCH.tar.gz && \
+    sudo tar -C /usr/local -xzvf go$VERSION.$OS-$ARCH.tar.gz && \
+    rm go$VERSION.$OS-$ARCH.tar.gz
+
+    # Download and install singularity from a release
+    export VERSION=3.0.3 && # adjust this as necessary \
+    mkdir -p $GOPATH/src/github.com/sylabs && \
+    cd $GOPATH/src/github.com/sylabs && \
+    wget https://github.com/sylabs/singularity/releases/download/v${VERSION}/singularity-${VERSION}.tar.gz && \
+    tar -xzf singularity-${VERSION}.tar.gz && \
+    cd ./singularity && \
+    ./mconfig
+
+    # Compile singularity
+    ./mconfig && \
+    make -C ./builddir && \
+    sudo make -C ./builddir install
+
+    singularity --version
```

Comment:
How long does this compilation take? 
It might be worth considering caching this https://docs.travis-ci.com/user/caching/

Reply:
Around 5 minutes for the whole install. will look into that

Reply:
I tried caching the singularity executable, but it has a bunch of dependencies which also had to be cached, and even after that I'm getting an error that I don't see any way to fix. Not sure this is possible or worth the effort.